### ISSUE TITLE
Quickstart transformation miscategorizes bill line

### ISSUE THREAD

**Issue by rishi-tripathy** [fivetran/dbt_quickbooks#105]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the issue

I am working with a customer to bring in Quickbooks data by GL.

I see a transaction the GENERAL_LEDGER transformation as being associated with Account 7410, but in Quickbooks it is clearly classified as 7620.

I believe this is because the inference of the account via the item number associated with the bill line is not happening or is happening incorrectly. The Bill itself is associated with account 7410, but the Bill Line in question is a specific item that should be categorized as 7620.

### Relevant error log or model output

```shell
As described above, there is a diff where more transactions that expected are being categorized in account 7410, since bill lines that are directly associated with an account are not being categorized as such
```

### Expected behavior

Use the `ACCOUNT_EXPENSE_ACCOUNT_ID` column on Bill Line table to associate bill line items with accounts as part of the transformations, instead of only relying on the parent bill. line items with direct associations to accounts should override their parent bill.

### dbt Project configurations

I'm using Fivetran's quickstart transformations for Quickbooks

### Package versions

I'm using Fivetran's quickstart transformations for Quickbooks

### What database are you using dbt with?

snowflake

### dbt Version

I'm using Fivetran's quickstart transformations for Quickbooks

### Are you willing to open a PR to help address this issue?

- [ ] Yes.
- [X] Yes, but I will need assistance and will schedule time during our [office hours](https://calendly.com/fivetran-solutions-team/fivetran-solutions-team-office-hours) for guidance
- [ ] No.

**Comment by fivetran-catfritz** [comment]
Hi @rishi-tripathy thank you for opening the issue--you bring up a very interesting case. I will be looking into this in the next week and will let you know if I have any questions!

**Comment by fivetran-catfritz** [comment]
Hi @rishi-tripathy, following up to my last message, are you seeing this issue in the credit or is this issue with the debit transactions?

**Comment by fivetran-catfritz** [comment]
Hi @rishi-tripathy have you had a chance to check into this? Thanks!

**Comment by rishi-tripathy** [comment]
Ah sorry for the delay here Cat! Notifications were piping to my personal email instead of my work one. I'm seeing the impact on debit items in this case. What's the implication of handing them separately?

**Comment by fivetran-catfritz** [comment]
Thanks @rishi-tripathy ! This is useful to narrow down where the issue might be. I had actually suspected credits, so it's very interesting the issue lies with debits. For debit transactions, we already use `account_expense_account_id` as the first option to determine `account_id` for debits (see [this line](https://github.com/fivetran/dbt_quickbooks/blob/main/models/double_entry_transactions/int_quickbooks__bill_double_entry.sql#L40) in our model), so I will have to investigate a bit more.

**Comment by fivetran-catfritz** [comment]
Hi @rishi-tripathy. I discussed this issue with my team, and it would be really helpful if you could show us live what is happening. Would you be available for a call in the next week? If so, please send an email to solutions@fivetran.com with your availability, and I'll set up the meet!

**Comment by fivetran-catfritz** [comment]
Hi @rishi-tripathy just following up if you would be available for a call with our team?

**Comment by rishi-tripathy** [comment]
Yes, can do cc @***@***.*** - how is this ( https://calendly.com/r-tripathy/30 ) availability?

Rishi Tripathy | He, Him, His
Twitter ( https://twitter.com/rishi_tripathy_ )
609-651-2559

**Comment by fivetran-catfritz** [comment]
Thanks @rishi-tripathy! I have scheduled a meeting.

**Comment by rishi-tripathy** [comment]
Great! Good timing as well - I just saw this issue pop up with another customer account.

The issue this time is with *debit* transaction types in *credit* account types. There are transactions included in the model (they're associated with the account number - in this case 4802) that should not be (i.e. they are not showing up as associated with that account in the customer's Quickbooks instance).

Happy to show you more live on our call tomorrow.

If you have a chance to poke around, two affected transactions from the QBO general_ledger transformation can be found by just filtering the entire transform model (org id is 709) by

adjusted_amount =-811 or =-292.40.

Rishi Tripathy | He, Him, His
Twitter ( https://twitter.com/rishi_tripathy_ )
609-651-2559

**Comment by rishi-tripathy** [comment]
In this second case (Would need to go back to confirm for the other case) — I can confirm that:

* the bill line in quickbooks.bill is associated with the same account (using `account_expense_account_id`) that it shows up in the model — though this is different than how it's categorized in the customer account. Asked the customer for more context on why this might be the case.

**Comment by fivetran-catfritz** [comment]
Hi @rishi-tripathy thanks for meeting with us on the call yesterday to go over this issue! To summarize what we found:
- The 7410 instead of 7620 account was associated with the bill line in question since in the source bill_line table, the `account_expense_account_id` did not have a value populated.
- This causes the coalesce clause in [this line](https://github.com/fivetran/dbt_quickbooks/blob/main/models/double_entry_transactions/int_quickbooks__bill_double_entry.sql#L40) to move on to another field to determine the account to use. 

#### Next steps:
- During the call we were unable to establish where the association of 7620 was coming from, so @rishi-tripathy will determine where that path is coming from. 

Keep us posted with what you find!
